frontend/src/components/Collaboration/CodeEditor.js

```diff
@@ -1,16 +1,30 @@
-import React, { useState, useEffect } from 'react';
+import React, { useState, useEffect, useRef } from 'react';
 import CodeMirror from '@uiw/react-codemirror';
 import { EditorView } from '@codemirror/view';
+import { vscodeDark } from "@uiw/codemirror-theme-vscode";
 import { python } from '@codemirror/lang-python';
 import { java } from '@codemirror/lang-java';
 import { cpp } from '@codemirror/lang-cpp';
 import { Language, Event } from '../../constants';
+import OverlayCursor from './OverlayCursor';
+import { isWithinWindow } from '../../utils/helpers';
 import '../../css/CodeEditor.css';
 
-const CodeEditor = ({ socket, roomId, selectedLanguage }) => {
+const CodeEditor = ({ socket, roomId, selectedLanguage, displayName, jwt }) => {
+  const editorBoxRef = useRef(null);
+
+  // Initialize code editor content
   const [code, setCode] = useState('');
   const [language, setLanguage] = useState(selectedLanguage);
 
+  // Initialize cursor position for code editor
+  const [scrollTop, setScrollTop] = useState(0);
+  const [scrollLeft, setScrollLeft] = useState(0);
+  const [position, setPosition] = useState({ x: 0, y: 0 });
+  const [opponent, setOpponent] = useState({});
```

Comment:
Overall LGTM. Just that the term 'opponent' sounds really weird. Perhaps the term 'partner' sounds better?

Reply:
Changed 👍